src/main/java/org/elasticsearch/action/trainnaivebayes/TransportTrainNaiveBayesAction.java

```diff
@@ -267,12 +270,25 @@ public Object run() {
                 }
             });
             String pmmlString = new String(outputStream.toByteArray(), Charset.defaultCharset());
-            client.prepareIndex(ScriptService.SCRIPT_INDEX, PMMLModelScriptEngineService.NAME, id).setSource("script", pmmlString)
-                    .execute(new ActionListener<IndexResponse>() {
+            if (id == null) {
+                //TODO: we can probably do better, but this should work for now
+                id = UUIDs.randomBase64UUID();
+            }
+            BytesReference source;
+            try {
+                XContentBuilder builder = XContentFactory.contentBuilder(Requests.INDEX_CONTENT_TYPE);
+                builder.startObject().field("script", pmmlString).endObject();
+                source = builder.bytes();
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            }
+            client.admin().cluster().preparePutStoredScript().setScriptLang(PMMLModelScriptEngineService.NAME)
+                    .setSource(source).setId(id)
+                    .execute(new ActionListener<PutStoredScriptResponse>() {
                         @Override
-                        public void onResponse(IndexResponse indexResponse) {
-                            listener.onResponse(new TrainNaiveBayesResponse(indexResponse.getIndex(), indexResponse.getType(),
-                                    indexResponse.getId()));
+                        public void onResponse(PutStoredScriptResponse indexResponse) {
+                            //TODO: Fix response
```

Comment:
That's means to remove index and type, which I did. But I forgot to remove TODO. Thanks for catching it.